telegram/ext/messagequeue.py

```diff
@@ -141,130 +181,189 @@ def stop(self, timeout: float = None) -> None:
+        """
+        Stops the all :class:`telegram.ext.DelayQueue` registered for this message queue.
 
-    stop.__doc__ = DelayQueue.stop.__doc__ or ''  # reuse docstring if any
+        Args:
+            timeout (:obj:`float`, optional): The timeout to pass to
+                :meth:`telegram.ext.DelayQueue.stop`.
+        """
+        self.running = False
+        for delay_queue in self._delay_queues.values():
+            delay_queue.stop(timeout)
 
-    def __call__(self, promise: Callable, is_group_msg: bool = False) -> Callable:
+    def put(self, func: Callable, delay_queue: str, *args: Any, **kwargs: Any) -> Promise:
         """
-        Processes callables in throughput-limiting queues to avoid hitting limits (specified with
-        :attr:`burst_limit` and :attr:`time_limit`.
+        Processes callables in throughput-limiting queues to avoid hitting limits.
 
         Args:
-            promise (:obj:`callable`): Mainly the ``telegram.utils.promise.Promise`` (see Notes for
-                other callables), that is processed in delay queues.
-            is_group_msg (:obj:`bool`, optional): Defines whether ``promise`` would be processed in
-                group*+*all* ``DelayQueue``s (if set to :obj:`True`), or only through *all*
-                ``DelayQueue`` (if set to :obj:`False`), resulting in needed delays to avoid
-                hitting specified limits. Defaults to :obj:`False`.
-
-        Note:
-            Method is designed to accept ``telegram.utils.promise.Promise`` as ``promise``
-            argument, but other callables could be used too. For example, lambdas or simple
-            functions could be used to wrap original func to be called with needed args. In that
-            case, be sure that either wrapper func does not raise outside exceptions or the proper
-            :attr:`exc_route` handler is provided.
+            func (:obj:`callable`): The callable to process
+            delay_queue (:obj:`str`): The name of the :class:`telegram.ext.DelayQueue` to use.
+            *args (:obj:`tuple`, optional): Arguments to ``func``.
+            **kwargs (:obj:`dict`, optional): Keyword arguments to ``func``.
 
         Returns:
-            :obj:`callable`: Used as ``promise`` argument.
+            :class:`telegram.ext.Promise`.
 
         """
+        return self._delay_queues[delay_queue].put(func, args, kwargs)
 
-        if not is_group_msg:  # ignore middle group delay
-            self._all_delayq(promise)
-        else:  # use middle group delay
-            self._group_delayq(self._all_delayq, promise)
-        return promise
+    def set_dispatcher(self, dispatcher: 'Dispatcher') -> None:
+        """
+        Sets the dispatcher to use for error handling.
+
+        Args:
+            dispatcher (:class:`telegram.ext.Dispatcher`): The dispatcher.
+        """
+        self.dispatcher = dispatcher
+
+    DEFAULT_QUEUE: ClassVar[str] = 'default_delay_queue'
```

Comment:
maybe add convenience properties `MQ.default/group_delay_queue` for accessing `MQ._delay_queue[MQ.DEFAULT/GROUP_QUEUE]`